distributed/s3.py

```diff
@@ -26,148 +32,526 @@
 
 get_s3_lock = threading.Lock()
 
+def split_path(path):
+    """
+    Normalise S3 path string into bucket and key.
+
+    Parameters
+    ----------
+    path : string
+        Input path, like `s3://mybucket/path/to/file`
+
+    Examples
+    --------
+    >>> split_path("s3://mybucket/path/to/file")
+    ("mybucket", "path/to/file")
+    """
+    if path.startswith('s3://'):
+        path = path[5:]
+    if '/' not in path:
+        return path, ""
+    else:
+        bits = path.split('/')
+        return bits[0], '/'.join(bits[1:])
```

Comment:
I think this can be `path.split('/', 1)`

```python
In [1]: '1,2,3,4,5'.split(',')
Out[1]: ['1', '2', '3', '4', '5']

In [2]: '1,2,3,4,5'.split(',', 1)
Out[2]: ['1', '2,3,4,5']
```